### PR TITLE
[F] Update HdCheckbox component to match the new design

### PR DESCRIPTION
**Pull request by leandroinacio**
https://homeday.atlassian.net/browse/DS-107

### CODE REVIEW

File: src/components/form/FieldBase.vue
Comment (metal-gogo) [inline src/components/form/FieldBase.vue:118]: I don't fully understand this computed property.

If we pass both the `name` and the `label`, then we trim the `name` and discard the `label`, right?

I think this could be simplified to:

```suggestion
    labelId() {
      return `${kebabCase(this.name.trim())}-label`;
    },
```

On one side, the label is only shown if the `label` prop is passed to the component.
We can consider that there will always be a `name` because it is marked as a required prop.

That's why I don't understand why we need to validate the presence of the `label` prop.

File: src/components/form/FieldBase.vue
Comment (viniciuskneves) [inline src/components/form/FieldBase.vue:113]: This is the change to tackle the `id` issue I mentioned in the other PR, right?

Ohhh, I just saw the `aria-labelledby` in the component. Now I got it hehehe
As it is something that caused some misunderstandings already, I would recommend to cover it with a test (HdCheckbox case).

File: src/components/form/FieldBase.vue
Comment (viniciuskneves) [reply src/components/form/FieldBase.vue:118]: I think the idea is to avoid having the `labelId` in the `aria-labelledby`, from `HdCheckbox`.

If that is the case, I would add a comment explaining it as it might be weird when revisiting this component in the future. As it is a single case, we don't have `aria-labelledby` in other inputs, it is a bit hard to reason the best structure for that. At the moment if fits our use case.

File: src/components/form/FieldBase.vue
Comment (leandroinacio) [reply src/components/form/FieldBase.vue:113]: Yep, that's it.
Unit test added 👍

**Review by volcanioo — APPROVED** [review]
Hey man, great job!

**Review by ilyasmez — APPROVED** [review]
Looks good to me 👍 Nice work!
Please feel free to merge once you get the design team's approval.

**Pull request merged**
